Neos.Fusion/Classes/FusionObjects/AbstractArrayFusionObject.php

```diff
@@ -130,27 +146,65 @@ protected function evaluateNestedProperties(?string $defaultFusionPrototypeName
      *
      * This will ignore all properties defined in "@ignoreProperties" in Fusion
      *
-     * @see PositionalArraySorter
-     *
      * @return array an ordered list of key value pairs
      * @throws FusionException if the positional string has an unsupported format
+     * @see PositionalArraySorter
+     *
+     * @deprecated
+     * @see preparePropertyKeys()
      */
     protected function sortNestedProperties(): array
     {
-        $arraySorter = new PositionalArraySorter($this->properties, '__meta.position');
+        return $this->preparePropertyKeys($this->properties);
+    }
+
+    /**
+     * @param array $properties
+     * @return array<string> Fusion keys in this Array fusion object
+     * @throws FusionException
+     */
+    protected function preparePropertyKeys(array $properties): array
+    {
+        $maybeSortedFusionKeys = $this->shouldSortProperties() ? $this->applyPositionalArraySorterToProperties($properties) : array_keys($properties);
+        return $this->filterIgnoredProperties($maybeSortedFusionKeys);
+    }
+
+    /**
+     * @param array $properties
+     * @return array
+     * @throws FusionException
+     */
+    protected function applyPositionalArraySorterToProperties(array $properties): array
+    {
+        $arraySorter = new PositionalArraySorter($properties, '__meta.position');
         try {
             $sortedFusionKeys = $arraySorter->getSortedKeys();
         } catch (InvalidPositionException $exception) {
             throw new FusionException('Invalid position string', 1345126502, $exception);
         }
 
+        return $sortedFusionKeys;
+    }
+
+    /**
+     * Filters propertyKeys by ignoredProperties
+     *
+     * Note: array_filter over propertyKeys might be more elegant but ignoredProperties will usually be the smaller set,
```

Comment:
indeed i wanted to change this wierd `for unset` loop once but found out about: https://gist.github.com/nunoveloso/1992851

seems phps inbuild functions are not always faster ...

Reply:
true, true, I actually refactored it now, this should definitely be fast(er).